homework5/5.1/5.1/Graph.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task5_1;
+
+/// <summary>
+/// Класс граф со списком его вершин,
+/// с матрицей смежности,
+/// с размером матрицы,
+/// с максимальным номером вершины
+/// </summary>
+public class Graph
+{
+    /// <summary>
+    /// Список вершин
+    /// </summary>
+    public List<Tuple<int, int, int>> Nodes { get; set; }
+
+    /// <summary>
+    /// Матрица смежности
+    /// </summary>
+    public int[,] Matrix { get; set; }
+
+    /// <summary>
+    /// Размер матрицы
+    /// </summary>
+    private int MatrixLength;
+
+    /// <summary>
+    /// Максимальный номер вершины
+    /// </summary>
+    private int MaxNumberNode;
+
+    /// <summary>
+    /// Инициализация графа
+    /// </summary>
+    /// <param name="filePath">Путь файла</param>
+    public Graph(string filePath)
+    {
+        Nodes = new List<Tuple<int, int, int>>();
+        MatrixLength = 20;
+        MaxNumberNode = 0;
+        Matrix = new int[MatrixLength, MatrixLength];
+        Parse(filePath);
+    }
+    /// <summary>
+    /// Функция, которая парсит файл
+    /// и представляет его данные в виде графа
+    /// </summary>
+    /// <param name="filePath">Путь файла</param>
+    private void Parse(string filePath)
+    {
+        using (StreamReader reader = new StreamReader(filePath))
```

Comment:
```suggestion
        using var reader = new StreamReader(filePath);
```
И дальше без фигурных скобок и отступа.